source/common/router/rds_impl.cc

```diff
@@ -64,13 +67,18 @@ StaticRouteConfigProviderImpl::~StaticRouteConfigProviderImpl() {
 RdsRouteConfigSubscription::RdsRouteConfigSubscription(
     const envoy::extensions::filters::network::http_connection_manager::v3::Rds& rds,
     const uint64_t manager_identifier, Server::Configuration::ServerFactoryContext& factory_context,
-    ProtobufMessage::ValidationVisitor& validator, Init::Manager& init_manager,
     const std::string& stat_prefix,
     Envoy::Router::RouteConfigProviderManagerImpl& route_config_provider_manager)
     : route_config_name_(rds.route_config_name()), factory_context_(factory_context),
-      validator_(validator), init_manager_(init_manager),
-      init_target_(fmt::format("RdsRouteConfigSubscription {}", route_config_name_),
-                   [this]() { subscription_->start({route_config_name_}); }),
+      validator_(factory_context.messageValidationContext().dynamicValidationVisitor()),
+      parent_init_target_(fmt::format("RdsRouteConfigSubscription init {}", route_config_name_),
+                          [this]() { local_init_manager_.initialize(init_watcher_); }),
+      init_watcher_(fmt::format("RDS local-init-watcher {}", rds.route_config_name()),
+                    [this]() { parent_init_target_.ready(); }),
```

Comment:
This makes sense to me and is roughly how I was envisaging transitive init managers working. @mergeconflict @lambdai did you have any thoughts here?

Reply:
LGTM. Based on my understanding this field initialization order doesn't introduce initialized-after-detroy

Reply:
I think it's fine, if it's needed.